write.go

```diff
@@ -38,3 +85,391 @@ func toOptSet(opts ...WriteOption) *writeOptSet {
+					w.WriteString(" ") // http://dicom.nema.org/medical/dicom/current/output/html/part05.html#sect_6.2
+				default:
+					w.WriteByte(0)
+			}
+		}
+		return nil
+}
+
+func writeBytes(w dicomio.Writer, values []byte, vr string) error {
+	var err error
+	if len(values) != 1 {
+		return fmt.Errorf("Expect a single value but found %v", values)
+	}
+	switch vr {
+		case "OW":
+			err = writeOtherWordString(w, values)
+		case "OB":
+			err = writeOtherByteString(w, values)
+		default:
+			return ErrorMismatchValueTypeAndVR
+		}
+		if err != nil {
+			return err
+		}
+		return nil
+	}
+
+func writeInts(w dicomio.Writer, values []int, vr string) error {
+	for _, value := range values {
+		switch vr {
+		case "US", "SS":
+			w.WriteUInt16(uint16(value))
+		case "UL", "SL":
+			w.WriteUInt32(uint32(value))
+		default:
+			return ErrorMismatchValueTypeAndVR
+		}
+	}
+	return nil
+}
+
+func writePixelData(w dicomio.Writer, t tag.Tag, value Value, vr string, vl uint32) error {
+	image := MustGetPixelDataInfo(value)
+	if vl == tag.VLUndefinedLength {
+		err := encodeElementHeader(w, t, vr, vl)
+		if err != nil {
+			return err
+		}
+		writeBasicOffsetTable(w, image.Offsets)
+		for _, frame := range image.Frames {
+				writeRawItem(w, frame.EncapsulatedData.Data)
+		}
+		err = encodeElementHeader(w, tag.SequenceDelimitationItem, "", 0)
+		if err != nil {
+			return err
+		}
+	} else {
+			numFrames := len(image.Frames)
+			numPixels := len(image.Frames[0].NativeData.Data)
+			numValues := len(image.Frames[0].NativeData.Data[0])
+			length := numFrames * numPixels * numValues * image.Frames[0].NativeData.BitsPerSample / 8 // length in bytes
+
+			err := encodeElementHeader(w, t, vr, uint32(length))
+			if err != nil {
+				return err
+			}
+			buf := new(bytes.Buffer)
+			buf.Grow(length)
+			for frame := 0; frame < numFrames; frame++ {
+				for pixel := 0; pixel < numPixels; pixel++ {
+					for value := 0; value < numValues; value++ {
+						if image.Frames[frame].NativeData.BitsPerSample == 8 {
+							binary.Write(buf, binary.LittleEndian, uint8(image.Frames[frame].NativeData.Data[pixel][value]))
+						} else if image.Frames[frame].NativeData.BitsPerSample == 16 {
+							binary.Write(buf, binary.LittleEndian, uint16(image.Frames[frame].NativeData.Data[pixel][value]))
+						}
+					}
+				}
+			}
+			w.WriteBytes(buf.Bytes())
+	}
+	return nil
+}
+
+// TODO implement
+func writeSequence(w dicomio.Writer, t tag.Tag, values []*SequenceItemValue, vr string, vl uint32, opts ...WriteOption) error {
+	return ErrorUnimplemented
+}
+
+// TODO implement
+func writeSequenceItem(w dicomio.Writer, t tag.Tag, values []*Element, vr string, vl uint32, opts ...WriteOption) error {
+	return ErrorUnimplemented
+}
+
+func writeOtherWordString(w dicomio.Writer, data []byte) error {
+	if len(data) % 2 != 0 {
+		return ErrorOWRequiresEvenVL
+	}
+	bo, _ := w.GetTransferSyntax()
+	r, err := dicomio.NewReader(bytes.NewBuffer(data), bo, int64(len(data)))
```

Comment:
it may be a little overkill to init a whole reader to read the data UInt16 at a time, but also not a huge deal for now, but maybe we should revisit.